app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -125,26 +123,9 @@ class RegisterController @Inject()(val sessionCacheConnector: SessionCacheConnec
     * been performed and the eligibility check is positive. This returns [[None]]
     * if all the above has successfully been performed and the eligibility check is negative.
     */
-  private def validateUser(authContext: AuthContext)(implicit hc: HeaderCarrier): Result[Option[UserInfo]] = for {
-    nino ← EitherT.fromOption[Future](retrieveNino(authContext), "Unable to retrieve NINO")
-    userInfo ← userInfoService.getUserInfo(authContext, nino)
-    eligible ← eligibilityConnector.checkEligibility(nino)
+  private def validateUser(userDetailsUri: String, nino: String)(implicit hc: HeaderCarrier): Result[Option[UserInfo]] = for {
```

Comment:
I know I wrote this, but `validateUser` can probably be called something better like `performEligibilityCheck` or something